test/DisintermediatedGrants.test.js

```diff
@@ -93,75 +80,47 @@ describe("DisintermediatedGrants", function () {
-            await whitelistDonor(this.grants, this.alice.address)
-            const donorBalance = await this.testERC20.balanceOf(this.alice.address)
-            await this.testERC20.connect(this.alice).approve(this.grants.address, 0)
-            await expect(
-                this.grants.connect(this.alice).donate(this.testERC20.address, donorBalance)
-            ).to.be.revertedWith("ERC20: transfer amount exceeds allowance")
-        })
-        it("can be made by whitelisted donors", async function () {
-            await whitelistDonor(this.grants, this.alice.address)
-            const donationCount = await this.grants.donationCount()
-            const tx = await this.grants.connect(this.alice).donate(this.testERC20.address, ERC20_TOKEN_AMOUNT)
-            const donation = await this.grants.donations(donationCount)
-            expect(donation.donor).to.equal(this.alice.address)
-            expect(donation.nativeToken).to.equal(false)
-            expect(donation.token).to.equal(this.testERC20.address)
-            expect(donation.amount).to.equal(ERC20_TOKEN_AMOUNT)
-            expect(donation.disbursedAmount).to.equal(0)
-            expect(donation.withdrawn).to.equal(false)
-            await expect(tx).to.emit(this.grants, "Donate").withArgs(donation)
-            expect(await this.testERC20.balanceOf(this.grants.address)).to.equal(ERC20_TOKEN_AMOUNT)
-        })
-        it("fail if donation amount is zero", async function () {
-            await whitelistDonor(this.grants, this.alice.address)
-            const donationCount = await this.grants.donationCount()
-            await expect(this.grants.connect(this.alice).donate(this.testERC20.address, 0)).to.be.revertedWith(
-                "donation amount cannot be zero"
+            await this.token.connect(this.alice).approve(this.grants.address, donationAmount)
+            await expect(this.grants.connect(this.alice).donate(this.token.address, donationAmount)).to.be.revertedWith(
+                "donation amount exceeds balance"
             )
         })
-    })
-    describe("native donations", function () {
-        it("cannot be made by non-whitelisted donors", async function () {
-            await expect(this.grants.connect(this.eve).donateNative({ value: ETH_AMOUNT })).to.be.revertedWith(
-                "caller is not whitelisted donor"
+        it("cannot be made with insufficient allowance", async function () {
+            await whitelistDonor(this.grants, this.alice.address)
+            const donorBalance = await this.token.balanceOf(this.alice.address)
+            await this.token.connect(this.alice).approve(this.grants.address, donorBalance.sub(1))
+            await expect(this.grants.connect(this.alice).donate(this.token.address, donorBalance)).to.be.revertedWith(
+                "insufficient donation amount allowance"
             )
         })
         it("can be made by whitelisted donors", async function () {
```

Comment:
Add assertion that indicates that Alice's balance hasn't changed (since we're no longer requiring donors to deposit any money into the contract)